EC2Connection should not ignore the host parameter

haridsv opened the pull request:
I noticed that the `host` parameter is being completely ignored in favor of what is coming from `config`.

danielgtaylor commented:
Would it be possible to add a unit test which shows the use case this is trying to fix? Thanks!

haridsv commented:
Should be possible, I will take a look at the test setup.

haridsv commented:
I added a unit test and verified that one of the tests fails without the fix. Apologies for not adding the test in the first place, didn't read through the CONTRIBUTING guidelines :)

haridsv commented:
@danielgtaylor is the unit test satisfactory?

haridsv commented:
For the sake of completeness, I am fixing other connection classes that have the same problem, so please hold on.

haridsv commented:
@danielgtaylor I have now searched and updated all the various connection classes to support the host parameter uniformly and also added unittests for each one of those. Please review the changes.
